### PR TITLE
Replace useContext( ViewContextContext ) with useViewContext.

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue:

- #5011.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/hooks/useViewContext.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useContext } from '@wordpress/element';
+import ViewContextContext from '../components/Root/ViewContextContext';
+
+/**
```
Comment (techanvil) [inline assets/js/hooks/useViewContext.js:25]: This comment block can be removed

File: assets/js/hooks/useViewContext.test.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+	it( 'should return the view context set in the nearest provider', () => {
+		const { result } = renderHook( () => useViewContext(), {
+			viewContext: VIEW_CONTEXT_DASHBOARD_VIEW_ONLY,
+		} );
+
+		expect( result.current ).toBe( VIEW_CONTEXT_DASHBOARD_VIEW_ONLY );
+	} );
+
+	const { result: dashboardResult } = renderHook( () => useViewContext(), {
```
Comment (techanvil) [inline assets/js/hooks/useViewContext.test.js:48]: Looks like this & the following lines may have been left in by accident, or are part of an unfinished test?

**Review by techanvil — CHANGES_REQUESTED** [review]
Nice one @tofumatt, looks good, I have left a couple of minor comments.

Also - having checked the branch out I've spotted a small handful of missed cases:

```bash
assets/js/modules/tagmanager/components/common/UseSnippetSwitch.js
40:     const viewContext = useContext( ViewContextContext );

assets/js/components/EntitySearchInput.js
63:     const viewContext = useContext( ViewContextContext );

assets/js/components/ModulesList.js
46:     const viewContext = useContext( ViewContextContext );

assets/js/components/DateRangeSelector.js
60:     const viewContext = useContext( ViewContextContext );

```

File: assets/js/hooks/useViewContext.test.js (hunk 2 above)
Comment (tofumatt) [reply assets/js/hooks/useViewContext.test.js:48]: Argh, whoops, they should have been in the test above but I must've messed up their placement. Fixed! 😅

**Comment by tofumatt** [comment]
Thanks! Changes addressed so I think this should be good now 🙂

**Review by techanvil — APPROVED** [review]
LGTM, nice one @tofumatt !

**Pull request merged**